src/averageAges.js

```diff
@@ -36,8 +49,29 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filteredWomen = people.filter(element => element.sex === 'f');
+
+  if (withChildren) {
+    const mother = people.map(element => element.mother);
+    const filteredMothers = filteredWomen.filter(elemenent =>
+      mother.includes(elemenent.name));
+    const ageWomenWithChildren = filteredMothers.map((person) =>
+      person.died - person.born);
+    const sumAgeWomenWithChildren = ageWomenWithChildren.reduce((age1, age2) =>
+      age1 + age2);
+    const avarageAgeWomenWithChildren = sumAgeWomenWithChildren
+    / ageWomenWithChildren.length;
+
+    return avarageAgeWomenWithChildren;
+  } else {
+    const ageWomen = filteredWomen.map((person) => person.died - person.born);
+    const sumAgeWomen = ageWomen.reduce((age1, age2) => age1 + age2);
+    const avarageAgeWomen = sumAgeWomen / ageWomen.length;
+
+    return avarageAgeWomen;
+  }
```

Comment:
your algorithm is not so concise, that can be. you can use only
const filteredWomen =  // and here with ternary operator create condition for filtering. the same with prev function
and then, move code that cal average age into separate method, and use it here and in prev function.